backend/gce.go

```diff
@@ -643,7 +643,11 @@ func (p *gceProvider) apiRateLimit(ctx gocontext.Context) error {
 		}
 
 		// Sleep for up to 1 second
+		var span *trace.Span
+		ctx, span = trace.StartSpan(ctx, "GCE.timeSleep.apiRateLimit")
 		time.Sleep(time.Millisecond * time.Duration(mathrand.Intn(1000)))
+		span.End()
```

Comment:
This one I think we need to do some `if trace.FromContext(ctx) != nil {` messiness to avoid orphaned spans.

The issue is that `apiRateLimit` is sometime called from outside the ProcessorRun root span, e.g. during setup. So we need to make sure to only add a child span if there actually is a parent.

The other trickiness here is that we need to make the `span.End()` conditional too, without using `defer`.